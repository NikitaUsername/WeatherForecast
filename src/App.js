import React from "react";
import Header from "./components/Header";
import Weather from "./components/Weather";
import 'bootstrap/dist/css/bootstrap.min.css';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import SavedHeader from "./components/SavedHeader";
import { connect } from 'react-redux';
import { handleModal } from './actions/savedCitiesActions'

class App extends React.Component {

  handleClose = () => {
    this.props.handleModal();
  };
  render() {
    return (
      <div>
        <Modal show={this.props.showModal} onHide={this.handleClose} animation={false}>
          <Modal.Header closeButton>
            <Modal.Title>Error</Modal.Title>
          </Modal.Header>
          <Modal.Body>{this.props.modalText}</Modal.Body>
          <Modal.Footer>
            <Button variant="primary" onClick={this.handleClose} >
              Ok
                        </Button>
          </Modal.Footer>
        </Modal>
        <Header />
        <Weather />
        <SavedHeader />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    showModal: state.showModal,
    modalText: state.modalText
  }
}
const mapDispatchToProps = {
  handleModal: handleModal
}

export default connect(mapStateToProps, mapDispatchToProps)(App);