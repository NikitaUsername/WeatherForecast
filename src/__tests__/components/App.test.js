import App from '../../App'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('App container', () => {
    test('matches the snapshot with city loader and no saved cities', () => {
        const mockGeolocation = {
            getCurrentPosition: jest.fn()
                .mockImplementation((success) => Promise.resolve(success({
                    coords: {
                        latitude: 10,
                        longitude: 10
                    }
                })))
        };
        navigator.geolocation = mockGeolocation;
        const store = mockStore({
            cities: [
            ]
        });
        const tree = renderer
            .create(
                <Provider store={store}>
                    <App />
                </Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot with city loader and one saved city', () => {
        const mockGeolocation = {
            getCurrentPosition: jest.fn()
                .mockImplementation((success) => Promise.resolve(success({
                    coords: {
                        latitude: 10,
                        longitude: 10
                    }
                })))
        };
        navigator.geolocation = mockGeolocation;
        const store = mockStore({
            cities: [
                {
                    name: 'Moscow',
                    weather: {}
                } 
            ]
        });
        const tree = renderer
            .create(
                <Provider store={store}>
                    <App />
                </Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});