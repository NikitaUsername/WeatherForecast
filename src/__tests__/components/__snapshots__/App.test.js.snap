// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container matches the snapshot with city loader and no saved cities 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4"
      >
        <h2
          id="title"
        >
          Погода здесь
        </h2>
      </div>
      <div
        className="col-lg-4"
      >
        <button
          id="update-button"
          onClick={[Function]}
        >
          Обновить
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <h2
          id="waiting"
        >
          Подождите, данные загружаются
        </h2>
      </div>
      <div
        className="col-lg-12 "
      >
        <div
          className="center-block"
        >
          <div
            className="css-236bqq"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4"
      >
        <h2
          id="title"
        >
          Избранное
        </h2>
      </div>
      <div
        className="col-lg-6"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            id="inputfld"
            placeholder="Город"
          />
          <input
            id="update-button"
            type="submit"
            value="Добавить в избранное"
          />
        </form>
      </div>
    </div>
    <div
      className="row"
    />
  </div>
</div>
`;

exports[`App container matches the snapshot with city loader and one saved city 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4"
      >
        <h2
          id="title"
        >
          Погода здесь
        </h2>
      </div>
      <div
        className="col-lg-4"
      >
        <button
          id="update-button"
          onClick={[Function]}
        >
          Обновить
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <h2
          id="waiting"
        >
          Подождите, данные загружаются
        </h2>
      </div>
      <div
        className="col-lg-12 "
      >
        <div
          className="center-block"
        >
          <div
            className="css-236bqq"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4"
      >
        <h2
          id="title"
        >
          Избранное
        </h2>
      </div>
      <div
        className="col-lg-6"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            id="inputfld"
            placeholder="Город"
          />
          <input
            id="update-button"
            type="submit"
            value="Добавить в избранное"
          />
        </form>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-12"
          >
            <h2
              id="waiting"
            >
              Подождите, данные загружаются
            </h2>
          </div>
          <div
            className="col-lg-12 "
          >
            <div
              className="center-block"
            >
              <div
                className="css-236bqq"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
